src/ctap2/attestation.rs

```diff
@@ -382,35 +412,44 @@ pub enum AttestationStatement {
 // }
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
-// See https://www.w3.org/TR/webauthn/#fido-u2f-attestation
+// See https://www.w3.org/TR/webauthn-2/#sctn-fido-u2f-attestation
+// u2fStmtFormat = {
+//                     x5c: [ attestnCert: bytes ],
+//                     sig: bytes
+//                 }
 pub struct AttestationStatementFidoU2F {
-    pub sig: Signature,
-
-    #[serde(rename = "x5c")]
     /// Certificate chain in x509 format
-    pub attestation_cert: Vec<AttestationCertificate>,
+    #[serde(rename = "x5c")]
+    pub attestation_cert: Vec<AttestationCertificate>, // (1) "x5c"
+    pub sig: Signature, // (2) "sig"
 }
 
-#[allow(dead_code)] // TODO(MS): Remove me, once we can parse AttestationStatements and use this function
 impl AttestationStatementFidoU2F {
     pub fn new(cert: &[u8], signature: &[u8]) -> Self {
         AttestationStatementFidoU2F {
-            sig: Signature(ByteBuf::from(signature)),
             attestation_cert: vec![AttestationCertificate(Vec::from(cert))],
+            sig: Signature(ByteBuf::from(signature)),
         }
     }
 }
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
-// TODO(baloo): there is a couple other options than x5c:
-//              https://www.w3.org/TR/webauthn/#packed-attestation
+// https://www.w3.org/TR/webauthn-2/#sctn-packed-attestation
+// packedStmtFormat = {
+//                       alg: COSEAlgorithmIdentifier,
+//                       sig: bytes,
+//                       x5c: [ attestnCert: bytes, * (caCert: bytes) ]
+//                   } //
+//                   {
+//                       alg: COSEAlgorithmIdentifier
+//                       sig: bytes,
+//                   }
 pub struct AttestationStatementPacked {
-    pub alg: COSEAlgorithm,
-    pub sig: Signature,
-
-    #[serde(rename = "x5c")]
+    pub alg: COSEAlgorithm, // (1) "alg"
+    pub sig: Signature,     // (2) "sig"
     /// Certificate chain in x509 format
-    pub attestation_cert: Vec<AttestationCertificate>,
+    #[serde(rename = "x5c", skip_serializing_if = "Option::is_none")]
+    pub attestation_cert: Option<Vec<AttestationCertificate>>, // (3) "x5c"
```

Comment:
One more thing, I think we want to use an empty vector here instead of None. See my previous comment here: https://github.com/mozilla/authenticator-rs/pull/214#issuecomment-1381060087.

Reply:
Oh, I missed that. Sorry for the confusion.